src/platforms/x11/x11_resources.cpp

```diff
@@ -101,6 +103,27 @@ class BasicXCBConnection : public mx::XCBConnection
         return atom;
     }
 
+    auto get_output_refresh_rate() const -> uint16_t override
+    {
+        // I'm assuming we handle xcb errors somewhere with events.
+        auto ver_cookie = xcb_randr_query_version_unchecked(conn, 1, 2);
+        xcb_randr_query_version_reply(conn, ver_cookie,nullptr);
+        auto screen_cookie = xcb_randr_get_screen_info_unchecked(conn,screen_->root);
+        auto screen_reply = xcb_randr_get_screen_info_reply(conn, screen_cookie, nullptr);
+        #ifdef __cpp_lib_format
+        auto log_msg = std::format("Detected {}Hz host output refresh rate.",screen_reply->rate);
+        #else
+        auto log_msg = [&]()
+        {
+          std::ostringstream x;
+          x << "Detected " << screen_reply->rate << "Hz host output refresh rate.";
+          return x.str();
+        }();
+        #endif
+        mir::logging::log(mir::logging::Severity::debug, log_msg, "mir:x11");
```

Comment:
```suggestion
        mir::log_debug("%s", log_msg.c_str());
```

Reply:
The existence of log_debug makes the whole thing  much much simpler.